### PR TITLE
Real time timers

### PR DESCRIPTION
**Pull request by MortonPL**
### Real time timers

- Timers can now display values in real time, taking game speed into account. This can be enabled with `RealTimeTimers=yes`.
- By default, time is calculated relative to desired framerate. Enabling `RealTimeTimers.Adaptive=yes` (always on for unlimited FPS) will calculate time relative to *current* FPS, accounting for lag.
  - When playing with unlimited FPS (or custom speed above 60 FPS), the timers might constantly change value because of the unstable nature.
- This option respects custom game speeds.

- This behavior is designed to be toggleable by users. For now you can only do that externally via client or manually.

In `ra2md.ini`:
```ini
[Phobos]
RealTimeTimers=no           ; boolean
RealTimeTimers.Adaptive=no  ; boolean
```

### CODE REVIEW

File: src/Misc/Hooks.Timers.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#include <Utilities/Macro.h>
+#include <GameOptionsClass.h>
+#include <FPSCounter.h>
+#include <SessionClass.h>
+#include <Phobos.h>
+
+int oldValue;
```
Comment (secsome) [inline src/Misc/Hooks.Timers.cpp:7]: NEVER pollute the namespace

Comment (MortonPL) [reply src/Misc/Hooks.Timers.cpp:7]: Then, is this okay?

```
namespace TimerValueTemp
{
	static int oldValue;
};
```

File: src/Misc/PhobosToolTip.cpp [hunk 2]
```diff
@@ -93,6 +94,19 @@ void PhobosToolTip::HelpText(BuildType& cameo)
...
 
+inline int tickTimeToSeconds(int tickTime)
+{
+	if (!Phobos::Config::RealTimeTimers)
+		return tickTime / 15;
+
+	if (Phobos::Config::RealTimeTimers_Adaptive
+		|| GameOptionsClass::Instance->GameSpeed == 0
+		|| (false && !SessionClass::IsMultiplayer())) // TODO change when custom game speed gets merged
+		return tickTime / std::max((int)FPSCounter::CurrentFrameRate, 1);
```
Comment (Metadorius) [inline src/Misc/PhobosToolTip.cpp:105]:
```suggestion
	{
		return tickTime / std::max((int)FPSCounter::CurrentFrameRate, 1);
	}
```

**Review by Metadorius — COMMENTED** [review]
How many frames does the FPS timer take into account? I think for adaptive game speed it would be beneficial to sample the framerate for a second or a few and then take the average frame rate for the last few seconds, otherwise the time will jitter *a lot*.

File: src/Misc/PhobosToolTip.cpp (hunk 2 above)
Comment (MortonPL) [reply src/Misc/PhobosToolTip.cpp:105]: `FPSCounter::CurrentFrameRate` comment says "The number of frames processed in the last second", which seems to be correct (?) when looking at decompiled code.

**Pull request merged**